src/OrchardCore/OrchardCore.Localization.Core/ServiceCollectionExtensions.cs

```diff
@@ -41,5 +43,35 @@ public static IServiceCollection AddPortableObjectLocalization(this IServiceColl
 
             return services;
         }
+
+        public static IServiceCollection AddDataAnnotationsPortableObjectLocalization(this IServiceCollection services)
+        {
+            Action<MvcDataAnnotationsLocalizationOptions> setupAction = options => {
+                var serviceProvider = services.BuildServiceProvider();
```

Comment:
i don't think it's a good pattern, better to use a transient option setup service (that will inject what it needs) registered through a module or our `OC.Mvc.Core startup`, but anyway after the mvc one, and then just use it to override the options set by mvc.

Reply:
There's an extension for `IMvcBuilder` I think, but it require `.AddMvc()`

Reply:
Agreed with @jtkech, don't build the service collection, ever.